mantidproject/mantid
# D2B feedback implementation

**Description of work.**

Would be ideal to have this in nightly by Monday (for an experiment), hence high priority.

This PR implements few features requested for D2B reduction, see the issue.

**Report to:** G.Cuello. 


**To test:**

- [ ] **Feature 1** Reducing component-wise

```
PowderDiffILLDetScanReduction(Run='535401.nxs', UseCalibratedData=False, Output2DTubes=True, Output1D=False,  CropNegativeScatteringAngles=False, OutputWorkspace='tube128',  InitialMask='0', FinalMask='0', ComponentsToReduce='tube_128')
PowderDiffILLDetScanReduction(Run='535401.nxs', UseCalibratedData=False, Output2DTubes=True, Output1D=False,  CropNegativeScatteringAngles=False, OutputWorkspace='alltubes',  InitialMask='0', FinalMask='0')
CropWorkspace(InputWorkspace='alltubes_2DTubes', OutputWorkspace='alltubes_tube128', XMin=147.475)
match = CompareWorkspaces(Workspace1='tube128_2DTubes', Workspace2='alltubes_tube128', Tolerance=0.000001)
assert(match[0])

```

- [ ] **Feature 2** When normalising to monitor, scale the counts up by 1M.

   In the example above, color fill plot of `all_2DTubes`. The color scale should be about 0-180.

- [ ] **Feature 3** Enabling/disabling the alignment

```
PowderDiffILLDetScanReduction(Run='535401.nxs', UseCalibratedData=False, Output2DTubes=True, CropNegativeScatteringAngles=False, OutputWorkspace='aligned',  InitialMask='0', FinalMask='0')
PowderDiffILLDetScanReduction(Run='535401.nxs', UseCalibratedData=False, Output2DTubes=True, CropNegativeScatteringAngles=False, OutputWorkspace='notaligned', AlignTubes=False, InitialMask='0', FinalMask='0')
``` 
Color fill plot of `aligned` and `notaligned` and confirm that in the `aligned` the zigzags seen in the top and bottom parts of `unaligned` are smoothened.

The rest are minor edits, captured by the system tests.

Crop the `.zip`, it's a nexus file.

[535401.nxs.zip](https://github.com/mantidproject/mantid/files/2353159/535401.nxs.zip)




Fixes #22832 . 

Does this update require release notes?


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
